packages/worker-shared/src/userAssetUploads.ts

```diff
@@ -26,43 +29,93 @@ export async function handleUserAssetUpload({
 
 export async function handleUserAssetGet({ request, bucket, objectName, context }: UserAssetOpts) {
 	const cacheUrl = new URL(request.url)
+	const shouldOptimize =
+		request.query.tl_opt && !/image-resizing/.test(request.headers.get('via') ?? '')
+
+	let format = null
+	if (shouldOptimize) {
+		const accept = request.headers.get('Accept')
+		if (accept && /image\/avif/.test(accept)) {
+			format = 'avif' as const
+		} else if (accept && /image\/webp/.test(accept)) {
+			format = 'webp' as const
+		}
+	}
+
+	// if we have a format, we want to cache the asset with the format in the URL
+	if (format) cacheUrl.searchParams.set('format', format)
```

Comment:
why does `format` matter for the cacheUrl? why not always cache?

Reply:
`format=auto` isn't supported by workers transforms

Reply:
two requests from different browsers might return differently formatted responses depending on what the browser `Accept` header contains. We can't cache an avif response and serve it to a browser that only supports jpg. This makes sure that we find and return a format in the cache that the requesting browser definitely supports

Reply:
looks like this was updated in the end 👍